mmv1/products/compute/terraform.yaml

```diff
@@ -1512,6 +1512,9 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           If set to `true`, default routes (`0.0.0.0/0`) will be deleted
           immediately after network creation. Defaults to `false`.
     properties:
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          regex: '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'
```

Comment:
drive-by-review: Should this call the `validateGCPName` function rather than introduce a new, subtly different regex?

Reply:
I think Prateek copied this based on your comment [here](https://github.com/hashicorp/terraform-provider-google/issues/9113#issuecomment-838839586) - so on the plus side, it's not a new regex :-) but +1 that using [validateGCPName](https://github.com/GoogleCloudPlatform/magic-modules/blob/bcc6331b8c8f79c723c458de2d74e9c733a4a27e/mmv1/third_party/terraform/utils/validation.go#L85) is probably better, assuming that's [compute-specific](https://cloud.google.com/compute/docs/naming-resources#resource-name-format) or there's other documentation showing it applies to this resource?

That would also remove the need to add a comment in the code to link to the documentation for the regex as part of this PR - though it would maybe be a good idea to _add_ a comment to that function as part of this PR.

Reply:
`validateGCPName` is GCE-specific, it got named at a point we pretty much only supported GCE resources. FWIW (and to my point about slightly different regexes) this is a slightly different regex than address- it's missing the ^ and $ to ensure it captures the entire string. I should have pointed to `validateGCPName` in the first place there, my bad on the change in opinion.

> That would also remove the need to add a comment in the code to link to the documentation for the regex as part of this PR - though it would maybe be a good idea to add a comment to that function as part of this PR.

+1

Reply:
> it's missing the ^ and $ to ensure it captures the entire string

oof, yeah, good catch. 👍

Reply:
Made the following changes-

1. Renamed validateGCPName to validateGCEName
2. Inside validateGCEName when error is thrown added link to naming convention https://cloud.google.com/compute/docs/naming-resources#resource-name-format